csi.proto

```diff
@@ -1316,6 +1331,13 @@ message NodeUnstageVolumeRequest {
   // system/filesystem, but, at a minimum, SP MUST accept a max path
   // length of at least 128 bytes.
   string staging_target_path = 2;
+
+  // Indicates that the SP should prefer to successfully unstage the
+  // volume, even if data loss would occur as a result.
+  // CO MUST NOT set this field to true unless SP has the
+  // FORCE_UNPUBLISH node capability.
+  // This in an OPTIONAL field.
+  bool force = 3 [(alpha_field) = true];
```

Comment:
I am not sure what driver will do differently with force is set to true or false.

Reply:
Force would suggest that a "umount -f" would be acceptable rather than a "umount". Generally umount will not succeed if buffered data would be lost because the underlying block device is unwritable. The -f flag tells umount to not worry about this and just kill the mount.
There can be similar checks at lower levels that might ordinarily fail if they can't be done "safely" (i.e. without losing data) that should be skipped in the presence of the force flag.

Reply:
It case it's not clear, the REASON for this is because if you've already cut off the node from the storage and moved the workload elsewhere (because the CO falsely detected the node was down and used the fence operation), then flushing the buffered data would result in corruption, so it's correct to discard it, but the SP cannot know this fact, so the flag gives the CO a way to communicate it. Ordinarily SPs are expected to fail unpublish/unstage if they can't be done safely because not failing in those cases could cause data corruption for a different reason.